### DISCUSSION TITLE
React Native | Controlled Field Array: Undefined Value

### DISCUSSION THREAD

**Discussion by natealmeida185** [react-hook-form/react-hook-form#7335] [General] [answered]
Hello, I am trying to create a dynamic TextInput where the user can append({name: ''") or add a pet name field for the user to fill out. I am unable to retrieve or read the user's input, the value console.log's as undefined. The result should be petname [{name: 'Dog1'}, {name: 'Dog2']

![react-native-form1](https://user-images.githubusercontent.com/70350322/146988078-0091bdf3-cf16-478e-8fca-d1c4a9e4f4a8.PNG)
![react-native-form2](https://user-images.githubusercontent.com/70350322/146988244-f15724e0-8d33-4ee8-b1a4-992b1dd25944.PNG)
![react-native-form3](https://user-images.githubusercontent.com/70350322/146988314-9a65adff-1815-4468-a5ff-f4dd297188d4.PNG)

**Comment by bluebill1049** [comment]
use Controller for react native controlled components: https://react-hook-form.com/api/usecontroller/controller

**Reply by natealmeida185** [reply]
How would I go about doing this in my example? Do I need onChange and value? Also, do I remove controlledFields in my mapping and just map through fields?

**Reply by bluebill1049** [reply]
have a look at this example: https://codesandbox.io/s/react-hook-form-usefieldarray-ssugn

**Reply by natealmeida185** [reply]
![image](https://user-images.githubusercontent.com/70350322/147019230-337bb222-32a8-419f-b237-b3efff597c4f.png)
![image](https://user-images.githubusercontent.com/70350322/147019170-93d8e524-39be-4728-9855-f3531a0fdcf6.png)
I am getting undefined values

**Reply by bluebill1049** [reply]
can you reproduce the issue on the above codesandbox?

**Comment by Moshyfawn** [comment, marked as answer]
[Here's a snack](https://snack.expo.dev/@moshyfawn/rn-controlled-inputs) with an example on how to use RN `<TextInput />` with RHF

**Reply by balloman** [reply]
This snack doesn't actually work???

**Comment by shubhambattoo** [comment]
Hi @Moshyfawn is your snack working on Android and iOS devices? as I am not able to get it to work with almost a similar codebase. Also your snack is not working on Android and iOS devices, I am not able to update the input values. I am able to update the input values when running in web mode.
